src/utils/getInitialURL.js

```diff
@@ -0,0 +1,46 @@
+export const parseUrl = () => {
+  const url = window.location.href
+  const geocoordmatch = url.substring(url.indexOf('@'))
+  console.log('match', geocoordmatch)
+
+  const getCoords = isValidCoord(geocoordmatch)
+
+  if (getCoords.valid) {
+    return {
+      center: { lat: getCoords.lat, lng: getCoords.lng },
+      zoom: getCoords.zoom,
+    }
+  } else {
+    return { center: { lat: 40.1125785, lng: -88.2287926 }, zoom: 1 }
+  }
+}
+
+const isValidCoord = (geocoordmatch) => {
+  const floatRegex = /^\-?[0-9]+(e[0-9]+)?(\.[0-9]+)?$/
+  console.log(geocoordmatch)
+  if (!geocoordmatch) {
+    return false
+  }
+  const param = geocoordmatch
+  //@lat,long,zoomz
+
+  const values = param.substring(1, param.length - 1).split(',')
```

Comment:
More specific name? Like `latlng`, `coordinateValues`, etc? But I think this contains the Z too, so those might not work. Also, perhaps make this an object so you can use like lat and lng down when you validate it instead of just values[0] and values[1] etc.